File: Day11/Challenge2.c
## Challenge
The program must accept two string values S1 and S2 containing only alphabets and an alphabet CH as the input.
The program must print the string which has the maximum occurrences of the CH. If both S1 and S2 have the same number of 
occurrences of the CH then the program must print the first string S1 as the output.
Note: At least one of the string values will always contain the alphabet CH.
Boundary Condition(s):
3 <= Length of S1, S2 <= 100
Input Format:
The first line contains S1.
The second line contains S2.
The third line contains CH.
Output Format:
The first line contains S1 or S2.
Example Input/Output 1:
Input:
world
hello
l
Output:
hello
Example Input/Output 2:
Input:
happy
day
a
Output:
happy

## Solution

#include<stdio.h>
#include<stdlib.h>
#include<string.h>

int main()
{
    char s1[100];
    char s2[100];
    char ch;
    scanf("%s %s %c", s1, s2, &ch);
    int c1=0;
    int c2=0;
    int len1= strlen(s1);
    int len2= strlen(s2);
    
    for(int i=0; i<len1; i++){
        if(s1[i]==ch){
            c1++;
        }
    }
    
    for(int i=0; i<len2; i++){
        if(s2[i]==ch){
            c2++;
        }
    }
    if(c1==c2){
        printf("%s", s1);
    }
    else{
        printf("%s", c1>c2? s1 : s2);
    }
}

File: Day25/Challenge2.java
//Challenge- Sum - Prime Integers ending with 7

/*
The program must accept an integer N as the input. The program must print the sum of prime
integers from 1 to N ending with 7 as the output.
Boundary Condition(s):
10 <= N <= 10^8
Input Format:
The first line contains N.
Output Format:
The first line contains the sum of prime integers from 1 to N ending with 7.
Example Input/Output 1:
Input:
42
Output:
61
Explanation:
The prime integers from 1 to 42 which ends with 7 are 7, 17 and 37.
The sum of 7, 17 and 37 is 61.
Hence the output is 61
Example Input/Output 2:
Input:
156
Output:
643
*/

//Solution

import java.util.*;

public class Hello {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        System.out.print(primeSumWith7(n));
    }


    public static boolean prime(int num) {
        if (num <= 1) {
            return false;
        }
        for (int i = 2; i * i <= num; i++) {
            if (num % i == 0) {
                return false;
            }
        }
        return true;
    }

    public static boolean contains7(int num) {
            if (num % 10 == 7) {
                return true;
            }
            return false;
    }
    public static int primeSumWith7(int n) {
        int sum = 0;
        for (int k = 2; k <= n; k++) {
            if (isPrime(k) && contains7(k)) {
                sum += k;
            }
        }
        return sum;
    }

}
